File: tests/fixtures/clean.py
# clean.py — fixture file for Python scanner integration tests.
# Each snippet is the safe equivalent of a pattern the scanner targets.
# The scanner should produce zero findings for this file.

import os
import subprocess
import hashlib
import requests

# Safe SQL: parameterised query — no injection risk
def get_user(conn, username):
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE name = %s", (username,))
    return cursor.fetchall()

# Safe subprocess: no shell=True, list form
def run_cmd(program, args):
    subprocess.run([program] + args, check=True)

# Safe eval alternative: ast.literal_eval for data only
import ast
def parse_literal(expr):
    return ast.literal_eval(expr)

# Safe hashing: SHA-256
def hash_password(password):
    return hashlib.sha256(password.encode()).hexdigest()

# Safe file read: sanitise the path first
def read_file(filename):
    safe_name = os.path.basename(filename)
    with open(os.path.join("/var/data", safe_name)) as f:
        return f.read()

# Safe HTTP: hardcoded trusted URL, no user input
def fetch_health():
    return requests.get("https://api.example.com/health", timeout=5)

File: tests/fixtures/clean.java
// clean.java — fixture file for Java scanner integration tests.
// Each snippet is the safe equivalent of a pattern the scanner targets.
// The scanner should produce zero findings for this file.

import java.sql.*;
import java.io.*;
import java.security.*;

public class CleanApp {

    // Safe SQL: parameterised query — no injection risk
    public ResultSet getUser(Connection conn, String username) throws SQLException {
        PreparedStatement stmt = conn.prepareStatement("SELECT * FROM users WHERE name = ?");
        stmt.setString(1, username);
        return stmt.executeQuery();
    }

    // Safe command execution: no string concatenation, fixed command
    public Process runCommand() throws IOException {
        ProcessBuilder pb = new ProcessBuilder("ls", "-la");
        return pb.start();
    }

    // Safe secret: loaded from environment
    private String apiKey = System.getenv("API_KEY");

    // Safe hashing: SHA-256
    public byte[] hashPassword(String password) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        return md.digest(password.getBytes());
    }

    // Safe file read: fixed path, no user input
    public String readConfig() throws IOException {
        File f = new File("/etc/app/config.json");
        return new String(java.nio.file.Files.readAllBytes(f.toPath()));
    }

    // Safe random: SecureRandom
    public String generateToken() {
        java.security.SecureRandom rng = new java.security.SecureRandom();
        return String.valueOf(rng.nextLong());
    }
}
